tests/functional/oauth_reset_password.js

```diff
@@ -159,12 +166,99 @@ registerSuite('oauth reset password', {
         }))
         .then(openVerificationLinkInSameTab(email, 0))
 
-        .then(testElementExists('#fxa-complete-reset-password-header'))
+        .then(testElementExists(selectors.COMPLETE_RESET_PASSWORD.HEADER))
         .then(fillOutCompleteResetPassword(PASSWORD, PASSWORD))
 
-        .then(testElementExists('#fxa-reset-password-complete-header'))
+        .then(testElementExists(selectors.RESET_PASSWORD_COMPLETE.HEADER))
         // user sees the name of the rp, but cannot redirect
         .then(testElementTextInclude('.account-ready-service', '123done'));
     }
+  },
+});
+
+
+registerSuite('oauth reset password with TOTP', {
+  beforeEach: function () {
+    // timeout after 90 seconds
+    this.timeout = TIMEOUT;
+    email = TestHelpers.createEmail();
+
+    return this.remote
+      .then(clearBrowserState({
+        '123done': true,
+        contentServer: true
+      }))
+      .then(createUser(email, PASSWORD, { preVerified: true }))
+      .then(fillOutSignIn(email, PASSWORD))
+      .then(testElementExists(selectors.SETTINGS.HEADER))
+      .then(enableTotp())
+      .then(_secret => {
+        secret = _secret;
+      })
+      .then(click(selectors.SETTINGS.SIGNOUT, selectors.SIGNIN.HEADER))
+      .then(clearBrowserState({
+        '123done': true,
+        contentServer: true
+      }))
+      .then(openFxaFromRp('email-first', { header: selectors.ENTER_EMAIL.HEADER }))
+      .then(type(selectors.ENTER_EMAIL.EMAIL, email))
+      .then(click(selectors.ENTER_EMAIL.SUBMIT, selectors.SIGNIN_PASSWORD.HEADER))
+
+      .then(click(selectors.SIGNIN_PASSWORD.LINK_FORGOT_PASSWORD))
+      .then(fillOutResetPassword(email));
+
+  },
+
+  tests: {
+    'reset password, verify same browser same tab': function () {
+      return this.remote
+        .then(openVerificationLinkInSameTab(email, 1))
+        .then(fillOutCompleteResetPassword(PASSWORD, PASSWORD))
+
+        .then(testElementExists(selectors.TOTP_SIGNIN.HEADER))
+        .then(type(selectors.TOTP_SIGNIN.INPUT, generateTotpCode(secret)))
+        .then(click(selectors.TOTP_SIGNIN.SUBMIT))
+
+        .then(testElementExists(selectors['123DONE'].AUTHENTICATED_TOTP));
+    },
+
+    'reset password, verify same browser different tab': function () {
+      return this.remote
+        .then(openVerificationLinkInNewTab(email, 1))
+        .then(switchToWindow(1))
+        .then(fillOutCompleteResetPassword(PASSWORD, PASSWORD))
+
+        .then(testElementExists(selectors.TOTP_SIGNIN.HEADER))
+        .then(type(selectors.TOTP_SIGNIN.INPUT, generateTotpCode(secret)))
+        .then(click(selectors.TOTP_SIGNIN.SUBMIT, selectors['123DONE'].AUTHENTICATED_TOTP));
```

Comment:
Can you continue this test to assert the original tab does *not* redirect or ask for TOTP?